test/Common/DurableClientBaseTests.cs

```diff
@@ -212,6 +228,30 @@ public void DurableClient_ExternalApp_CreateHttpManagementPayload_ThrowsExceptio
             Assert.ThrowsAny<InvalidOperationException>(() => durableOrchestrationClient.CreateHttpManagementPayload("testInstanceId"));
         }
 
+        /// <summary>
+        /// End to end test that ensures that DurableClientFactory is set up correctly
+        /// (i.e. the correct services are injected through dependency injection
+        /// and AzureStorageDurabilityProvider is created).
+        /// </summary>
+        [Fact]
+        [Trait("Category", PlatformSpecificHelpers.TestCategory)]
+        public async Task DurableClient_AzureStorage_SuccessfulSetup()
+        {
+            string orchestratorName = nameof(TestOrchestrations.SayHelloInline);
+            using (ITestHost host = TestHelpers.GetJobHost(
```

Comment:
In the long run, we may be able to simplify this test (and the additional helper methods we added) by making a generic .NET Core host that calls `builder.Services.AddDurableClientFactory();`. That is more inline with what end customers would be doing, and then you don't need all of the changes to TestHelpers to accommodate wiring this into the JobHost.

In the interest of time and the code freeze today, I am fine postponing this.